core/lib/zksync_core/src/commitment_generator/mod.rs

```diff
@@ -0,0 +1,264 @@
+                let bootloader_initial_content_commitment = bootloader_initial_content_commitment(
+                    &initial_bootloader_contents,
+                    protocol_version,
+                )
+                .context("Bootloader content commitment is required for post-boojum batch")?;
+                latency.observe();
+
+                Ok(bootloader_initial_content_commitment)
+            });
+
+        let events_queue_commitment = events_commitment_task
+            .await
+            .context("`events_commitment_task` failed")??;
+        let bootloader_initial_content_commitment = bootloader_memory_commitment_task
+            .await
+            .context("`bootloader_memory_commitment_task` failed")??;
+
+        Ok(AuxCommitments {
+            events_queue_commitment,
+            bootloader_initial_content_commitment,
+        })
+    }
+
+    async fn prepare_input(
+        &self,
+        l1_batch_number: L1BatchNumber,
+    ) -> anyhow::Result<CommitmentInput> {
+        let mut connection = self
+            .connection_pool
+            .access_storage_tagged("commitment_generator")
+            .await?;
+        let header = connection
+            .blocks_dal()
+            .get_l1_batch_header(l1_batch_number)
+            .await?
+            .context("header is missing for batch")?;
+        let tree_data = connection
+            .blocks_dal()
+            .get_l1_batch_tree_data(l1_batch_number)
+            .await?
+            .context("tree data is missing for batch")?;
+
+        // TODO(PLA-731): ensure that the protocol version is always available.
+        let protocol_version = header
+            .protocol_version
+            .unwrap_or_else(ProtocolVersionId::last_potentially_undefined);
+        let common = CommitmentCommonInput {
+            l2_to_l1_logs: header.l2_to_l1_logs,
+            rollup_last_leaf_index: tree_data.rollup_last_leaf_index,
+            rollup_root_hash: tree_data.hash,
+            bootloader_code_hash: header.base_system_contracts_hashes.bootloader,
+            default_aa_code_hash: header.base_system_contracts_hashes.default_aa,
+            protocol_version,
+        };
+        let touched_slots = connection
+            .storage_logs_dal()
+            .get_touched_slots_for_l1_batch(l1_batch_number)
+            .await?;
+        let touched_hashed_keys: Vec<_> =
+            touched_slots.keys().map(|key| key.hashed_key()).collect();
+        let previous_values = connection
+            .storage_logs_dal()
+            .get_previous_storage_values(&touched_hashed_keys, l1_batch_number)
+            .await?;
+        let l1_batches_for_initial_writes = connection
+            .storage_logs_dal()
+            .get_l1_batches_and_indices_for_initial_writes(&touched_hashed_keys)
+            .await?;
+        drop(connection);
+
+        let input = if protocol_version.is_pre_boojum() {
```

Comment:
Nit: IMO, this `prepare_input` method is uncomfortably large. It would be easier to read if computing `CommitmentInput` for pre-Boojum / post-Boojum batches was extracted to separate methods.

Reply:
I tried separating it, but to avoid unnecessary code duplication for loading data from DB, methods for pre-boojum and post-boojum input prep should take lots of args and it doesn't look nice as well, so I decided to keep it this way